File: views/categories/map.js
function(item) {// we create a copy of the item to operate on.
	// !code lib/jquery/core.js
	
	var holder = {};
	var keycodes = item['keycodes'];
	var presentations = item['presentations'];
	
	// copy item into the holder array
	$.extend(true,holder,item);
	
	// remove the keys from the temporary item
	delete holder['keycodes'];
	delete holder['presentations'];

	var p_list = {};
	
	var section = holder['type']+'s';

	fill_presentations = function(code,presentation_list,type) {
		for(var key in presentation_list) {
			presentation = presentation_list[key];
				if(presentation.type == 'category') {
					for(var override_key in presentation) {
						if(override_key != 'type') {
							holder[override_key] = presentation[override_key];
						}
					}
					var value = {};
					var temp = {};
					value[section] = [];
					value[section].push(holder);
					emit([key,code],value);
				}
			}
		};
	
	fill_presentations('DEFAULT',item.presentations);
	
	// okay, just ignore erros for now.
	try{
		$.each(keycodes,function(v){
			fill_presentations(v.code,v.presentations);
		});
	}
	catch(err){;}
}

File: views/category/map.py
def fun(item): 
    
    holder = {}
    keycodes = item['keycodes']
    
    master_item = item.copy()
    
    del master_item['keycodes']
    del master_item['presentations']
    
    holder = master_item.copy()

    keycodes = item['keycodes']
    # move any presentation not under a "keycode" into the "DEFAULT" keycode
    keycodes['DEFAULT'] = item['presentations']
    del item['presentations']
    
    for keycode in item['keycodes']:
        for presentation in keycode['presentations'].keys():
            # restore original item settings for each presentation
            holder = master_item.copy()
            if presentation['type'] == 'category':
                for attribute in presentation.keys():
                    if attribute <> 'type':
                        holder[attribute] = presentation[attribute]
                yield [presentation.key, keycode], [holder[x] for x in holder.keys()]